pkg/sideinputs/initializer/initializer.go

```diff
@@ -46,15 +53,21 @@ func NewSideInputsInitializer(isbSvcType dfv1.ISBSvcType, pipelineName, sideInpu
 }
 
 func (sii *sideInputsInitializer) Run(ctx context.Context) error {
+	var (
+		isbSvcClient isbsvc.ISBService
+		natsClient   *jsclient.NATSClient
+		err          error
+	)
+
 	log := logging.FromContext(ctx)
 	log.Infow("Starting Side Inputs Initializer", zap.Strings("sideInputs", sii.sideInputs))
 
-	var isbSvcClient isbsvc.ISBService
 	switch sii.isbSvcType {
 	case dfv1.ISBSvcTypeRedis:
 		return fmt.Errorf("unsupported isbsvc type %q", sii.isbSvcType)
 	case dfv1.ISBSvcTypeJetStream:
-		natsClient, err := jsclient.NewNATSClient(ctx)
+		natsClient, err = jsclient.NewNATSClient(ctx)
+		defer natsClient.Close()
 		if err != nil {
 			log.Errorw("Failed to get a NATS client.", zap.Error(err))
 			return err
```

Comment:
close the client only when the client creation is successful, if it fails we will get a seg fault 
move `defer natsClient.Close()` to line 75